crown-court-contribution/src/test/java/uk/gov/justice/laa/crime/contribution/service/ContributionServiceTest.java

```diff
@@ -0,0 +1,218 @@
+package uk.gov.justice.laa.crime.contribution.service;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.justice.laa.crime.contribution.common.Constants;
+import uk.gov.justice.laa.crime.contribution.data.builder.TestModelDataBuilder;
+import uk.gov.justice.laa.crime.contribution.dto.AssessmentRequestDTO;
+import uk.gov.justice.laa.crime.contribution.dto.AssessmentResponseDTO;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+@ExtendWith(MockitoExtension.class)
+class ContributionServiceTest {
```

Comment:
This test class can be significantly simplified by using parameterised unit tests.
If you are unsure on how best to do this or if you haven't used parameterised tests before do let me know, am more than happy to pair with you on it.

https://www.baeldung.com/parameterized-tests-junit-5

Reply:
That's looking a bit better, with less duplication!
To make it easier to read (and fit on one screen), I would consider using static imports for the `Constants` like this:

`import static uk.gov.justice.laa.crime.contribution.common.Constants.*;
`

i.e.
```
package uk.gov.justice.laa.crime.contribution.service;

import org.junit.jupiter.api.extension.ExtendWith;
....
import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
import static uk.gov.justice.laa.crime.contribution.common.Constants.*;

@ExtendWith(MockitoExtension.class)
class ContributionServiceTest {
    private static final String PASSPORT_RESULT_FAIL_CONTINUE = TestModelDataBuilder.PASSPORT_RESULT_FAIL_CONTINUE;

    @InjectMocks
    private ContributionService contributionService;

    private static Stream<Arguments> getAssessmentRequestForIojResult() {
        return Stream.of(
                Arguments.of(new AssessmentRequestDTO(PASS, PASS, null, PASS, FULL, PASS), PASS),
                Arguments.of(new AssessmentRequestDTO(PASS, null, null, PASS, null, PASS), PASS)
        );
    }

    private static Stream<Arguments> getAssessmentRequestForMeansResult() {
        return Stream.of(
                Arguments.of(new AssessmentRequestDTO(PASS, PASS, null, PASS, FULL, PASS), FULL),
                Arguments.of(new AssessmentRequestDTO(PASS, null, null, PASS, null, PASS), INIT.concat(PASS)),
                Arguments.of(new AssessmentRequestDTO(PASS, null, null, null, null, PASS), NONE),
                Arguments.of(new AssessmentRequestDTO(PASS, PASS, PASS, PASS, FULL, PASS), PASSPORT),
                Arguments.of(new AssessmentRequestDTO(PASS, PASS, FAIL, PASS, FULL, PASS), FAILPORT),
                Arguments.of(new AssessmentRequestDTO(PASS, PASS, PASSPORT_RESULT_FAIL_CONTINUE, PASS, PASS, PASS), PASS),
                Arguments.of(new AssessmentRequestDTO(PASS, PASS, PASSPORT_RESULT_FAIL_CONTINUE, FAIL, PASS, PASS), PASS),
                Arguments.of(new AssessmentRequestDTO(PASS, PASS, PASSPORT_RESULT_FAIL_CONTINUE, FAIL, FAIL, PASS), PASS),
                Arguments.of(new AssessmentRequestDTO(PASS, PASS, PASSPORT_RESULT_FAIL_CONTINUE, FAIL, FAIL, FAIL), FAIL),
                Arguments.of(new AssessmentRequestDTO(PASS, PASS, PASSPORT_RESULT_FAIL_CONTINUE, FAIL, FAIL, null), FAIL),
                Arguments.of(new AssessmentRequestDTO(PASS, PASS, PASSPORT_RESULT_FAIL_CONTINUE, FULL, FAIL, FAIL), FAIL),
                Arguments.of(new AssessmentRequestDTO(PASS, PASS, PASSPORT_RESULT_FAIL_CONTINUE, FULL, FAIL, null), FAIL),
                Arguments.of(new AssessmentRequestDTO(PASS, PASS, PASSPORT_RESULT_FAIL_CONTINUE, HARDSHIP_APPLICATION, FAIL, FAIL), FAIL),
                Arguments.of(new AssessmentRequestDTO(PASS, PASS, PASSPORT_RESULT_FAIL_CONTINUE, HARDSHIP_APPLICATION, FAIL, null), FAIL)

        );
    }
....
}
```

Reply:
Fixed now